msal/src/main/java/com/microsoft/identity/client/AccountAdapter.java

```diff
@@ -51,16 +51,27 @@ private interface CacheRecordFilter {
     }
 
     private static class GuestAccountFilter implements CacheRecordFilter {
+        final static String TAG = GuestAccountFilter.class.getSimpleName();
 
         @Override
         public List<ICacheRecord> filter(@NonNull List<ICacheRecord> records) {
+            final String methodTag = TAG + ":filter";
+
             final List<ICacheRecord> result = new ArrayList<>();
 
             for (final ICacheRecord cacheRecord : records) {
                 final String acctHomeAccountId = cacheRecord.getAccount().getHomeAccountId();
                 final String acctLocalAccountId = cacheRecord.getAccount().getLocalAccountId();
-
-                if (!acctHomeAccountId.contains(acctLocalAccountId)) {
+                boolean notNullorEmpty = true;
+                if (StringUtil.isNullOrEmpty(acctHomeAccountId)) {
+                    Logger.warn(methodTag, "Home account id is null or empty.");
```

Comment:
In practice, how often does this happen? I'm just asking because these are `warn` level and I don't want to flood the logs with too much of these statements in cases when these being null is expected behavior

Reply:
We don't know yet. Based on how the original code was written, it doesn't seem that either home id or local account id were expected to be null, but I also think that it's possible different scenarios have been created since then. The issue is still being actively investigated, so when I get more info, I'll make another PR to remove the logging if we need to consider null as expected.